BatchCave.py

```diff
@@ -18,31 +26,41 @@
 
     **** UC Libraries Record Loads Working Group ****
 
-"""
+    """)
 
 def BrowseFiles():#open file explorer
-    root=Tk()
+    root=tk.Tk()
     root.withdraw()
     filenameOUT = askopenfilename(filetypes=[("MARC files","*.mrc"),("XML files","*.xml"),("All the files","*.*")], title="R.L.W.G Batch Edit -- select input file")
-    print type(filenameOUT)
+    print (type(filenameOUT))
     filename = filenameOUT
     
     root.destroy()
-    print '\n\nSelected file: \"' + re.sub('.*/(.*?)$', '\\1\"', filename) 
+    print ('\n\nSelected file: \"' + re.sub('.*/(.*?)$', '\\1\"', filename) )
 
     return filename
 
 class utilityFunctions:
 
+
     def listChangeScripts(self, BatchEdits):
-        
-        num = 0
-        ChangeScriptsDict = {}
-        for i in dir(BatchEdits)[:-2]:
-            num = num + 1
-            ChangeScriptsDict[num] = [i, ''.join(inspect.getargspec(getattr(BatchEdits, i))[3])]
+        num = range(len(dir(BatchEdits)[:-26]))

```

Comment:
Can you add a comment to document what the `[:-26]` slice is doing here?

Reply:
Okay, so when we print out the list of functions to change the script the names should be printed, and they are but all a list of 26 other attributes like "self" is printed. So I counted the number of instances that happened and then double checked by counting the number of modules and comparing it to how many should be listed. Then decided to hard code -26, so it left off those attributes its reading from the function declarations.